server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -217,68 +210,93 @@ public void restoreFromRemoteStore(RestoreRemoteStoreRequest request, final Acti
+                        if (restoreAllShards && IndexMetadata.State.CLOSE.equals(indexMetadata.getState())) {
+                            String errorMsg = "cannot restore index ["
+                                + indexName
+                                + "] because an open index "
+                                + "with same name already exists in the cluster. Close the existing index";
+                            if (allowPartial) {
+                                throw new IllegalStateException(errorMsg);
+                            } else {
+                                logger.warn(errorMsg);
                             }
-                            updatedIndexMetadata = IndexMetadata.builder(currentIndexMetadata)
-                                .state(IndexMetadata.State.OPEN)
-                                .version(1 + currentIndexMetadata.getVersion())
-                                .mappingVersion(1 + currentIndexMetadata.getMappingVersion())
-                                .settingsVersion(1 + currentIndexMetadata.getSettingsVersion())
-                                .aliasesVersion(1 + currentIndexMetadata.getAliasesVersion())
-                                .build();
-                        } else {
-                            activeInitializingShards = currentState.routingTable()
-                                .index(index)
-                                .shards()
-                                .values()
-                                .stream()
-                                .map(IndexShardRoutingTable::primaryShard)
-                                .filter(shardRouting -> shardRouting.unassigned() == false)
-                                .collect(Collectors.toMap(ShardRouting::shardId, Function.identity()));
                         }
+                        // other validation will come here
+                    } else {
+                        logger.warn("Remote store is not enabled for index: {}", indexName);
+                    }
 
-                        IndexId indexId = new IndexId(index, updatedIndexMetadata.getIndexUUID());
+                }
+            }
 
-                        RemoteStoreRecoverySource recoverySource = new RemoteStoreRecoverySource(
-                            restoreUUID,
-                            updatedIndexMetadata.getCreationVersion(),
-                            indexId
-                        );
-                        rtBuilder.addAsRemoteStoreRestore(updatedIndexMetadata, recoverySource, activeInitializingShards);
-                        blocks.updateBlocks(updatedIndexMetadata);
-                        mdBuilder.put(updatedIndexMetadata, true);
-                        indicesToBeRestored.add(index);
-                        totalShards += updatedIndexMetadata.getNumberOfShards();
+            private ClusterState executeRestore(ClusterState currentState, Map<String, IndexMetadata> indexMetadataMap,
```

Comment:
on a high level restore code remains as is, just moved around some stuff to separate out validation and actual restore